src/Cake/Utilities/FastHash.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Cake.Utilities
+{
+    /// <summary>
+    /// Optimized hash generator. Using SHA1CryptoServiceProvider since it is FIPS compliant.
+    /// </summary>
+    internal static class FastHash
+    {
+        /// <summary>
+        /// Generates a hash of the passed byte arrays.
+        /// </summary>
+        /// <param name="input">The binary data to hash.</param>
+        /// <returns>The hash value.</returns>
+        public static string GenerateHash(byte[] input)
+        {
+            using (var sha1 = new SHA1CryptoServiceProvider())
```

Comment:
Should we use `SHA256` or `SHA512`?

Reply:
Yes, `sha1` will cause compliance issues,  also providers are now obsolete so the base Create method must be used.

I've pushed a commit that uses `SHA512` and new more performant `Convert.ToHexString` on .NET 5 / 6.